server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -90,7 +92,11 @@ public FastenKafkaPlugin(boolean enableKafka, Properties consumerProperties, Pro
         this.plugin = plugin;
 
         if (enableKafka) {
-            this.connection = new KafkaConsumer<>(consumerProperties);
+            this.connNorm = new KafkaConsumer<>(consumerProperties);
+            // For priority connection, the client should be different from the normal one
+            Properties consumerPrioProperties = (Properties) SerializationUtils.clone(consumerProperties);
```

Comment:
Please factor out a helper method `cloneWithNewId(p: Properties, prefix: String) : Properties` that clones he properties and sets a new id by appending a provided prefix to the old one.

Reply:
I've created a private method to do this. Thanks for the comment.